### PR TITLE
Require payload member types to be string or number.

### PR DESCRIPTION
**Pull request by bharathcs**
- Set type standard for payloads.
- Update `BaseApi`.
- Update api/app/[:appId]/tasks
- Update stages type.
- Update api/app/:appId/stages
- Update api/app/:appId/stages/:stageId.
- Refactor to have a single converter.
- Create TaskData converter.
- Update api/applications.
- Put in explicit types for handlers.

### CODE REVIEW

File: utils/task/converter.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { Task } from '@prisma/client';
+import { TaskData } from '../../types/task';
+
+export function convertTaskToPayload(task: Task): TaskData {
+  const { id, title, dueDate, notificationDateTime, isDone } = task;
+  return {
+    id,
+    title,
+    dueDate: dueDate.toJSON() ?? null,
```
Comment (zognin) [inline utils/task/converter.ts]: I don't think dueDate is supposed to be null?
```suggestion
    dueDate: dueDate.toJSON(),
```

File: types/applicationStage.ts [hunk 2]
```diff
@@ -28,6 +28,6 @@ export type ApplicationStagePatchData = {
 export type ApplicationStageApplicationListData = {
   id: number;
   type: ApplicationStageType;
-  date: Date;
+  date: string;
   emojiUnicodeHex: Nullable<string>;
 };
```
Comment (zognin) [inline types/applicationStage.ts:33]: I think we wanted to change post and patch data date types to string too? `ApplicationStagePatchData` and `ApplicationStagePostData` weren't changed yet.

File: pages/api/applications/index.ts [hunk 3]
```diff
@@ -240,4 +216,30 @@ function validatePostRequest(req: NextApiRequest): Nullable<MessageType> {
...
+    id,
+    taskNotificationCount,
+    role: {
+      type,
+      title,
+      year,
+      id: role.id,
+      company: { id: company.id, name: company.name, companyUrl: company.companyUrl },
+    },
+    latestStage: convertApplicationStageToPayload(latestStage),
```
Comment (zognin) [inline pages/api/applications/index.ts]: `latestStage` may be undefined if an application has no stages.
Screenshot 2022-09-16 at 12 29 35 PM

**Review by zognin — COMMENTED** [review]
Mostly looks good thanks for doing this

File: utils/task/converter.ts (hunk 1 above)
Comment (bharathcs) [reply utils/task/converter.ts]: Oh, thanks for the catch!

**Review by zognin — APPROVED** [review]
LGTM thanks!

**Pull request merged**